pex/bin/pex.py

```diff
@@ -892,6 +892,27 @@ def build_pex(
         "Adding distributions from pexes: {}".format(" ".join(options.requirements_pexes))
     ):
         for requirements_pex in options.requirements_pexes:
```

Comment:
Can the Pants-aware reviewers comment on why this option is used by Pants at all? Its an odd duck option in the PEX cli and smells like a hack for Pants sake, but I can't recall exactly why this was deemed necessary. Since PEX_PATH nets you the same thing, but not as a single file, I suspect this must have to do with creating a single file PEX more quickly than it might otherwise be created? I'd love to know a bit more to help evaluate if this can be axed for Pex 3.

Reply:
Pants uses this only to support "local dists" - which is how it supports building (unversioned, unpublished) local native code into wheels (via setup.py) and then packaging those into the final .pex file along with local Python code. 

To [quote you](https://github.com/pantsbuild/pants/issues/13587#issuecomment-968160031): "The local PEX is a hack and can be "broken" since its never exposed to or used by end users. It's just there to emulate building a wheel for each local dist and then adding that wheel to another PEX."

So some alternative way to add unpublished wheels directly to a pex would be fine. There is more context in [that issue](https://github.com/pantsbuild/pants/issues/13587#issuecomment-969088777) about the perf reasons that led to the current solution, and you have suggested an alternative there that Pants could adopt and is more robust, it just takes some work.

Reply:
Ok, thanks for the pointers. I'll have to re-read. Clearly though PEX already supports this, you `-f dir/` for the dir containing the unpublished wheel and just do a normal resolve, lock, what have you.

Reply:
Yeah @benjyw, so after re-reading, Pants gets itself into trouble by not performing a normal resolve. If it's at all possible to:
1. Build the wheel
2. Do the resolve with either `pex -f find-links/ ...` or else `pex {normal reqs...} {wheel1} {wheel2} ... {wheeln) ...`.

That is the sane thing to do, IOW treat the wheels Pants builds like any other with no PEX hacks that attempt to stitch together resolves post-facto. In general, and this includes PEX_PATH, any attempt to cobble together a sys.path outside of running a single unified resolve is asking for all sorts of trouble as a user and all sorts of complication for Pex to support since Pex is not a resolver, it just uses one.